### PR TITLE
Add AI usage guidance document for Tidy Dev Day

### PR DESCRIPTION
**Pull request by mine-cetinkaya-rundel**

### CODE REVIEW

**Review by jennybc — APPROVED** [review]
I like it!

It resonates with some things we'll also say in our workshop. I'm going to mention this blog post: [Code is the byproduct](https://yagmin.com/blog/code-is-the-byproduct/). At the end, it highlights some stuff that's very compatible with what's here. Adding in case we want to recommend they read it in advance or borrow some language.

* When planning a feature:

  Can this feature reuse any existing code? Is there an existing pattern the design of this feature should follow? Describe the architecture of this feature. Where are the bottlenecks? Is this an extensible approach if I want to add Feature B down the road? Is there a way to implement this without affecting this other part of the codebase?
* When writing/generating code:

  Does this consider edge cases along these lines? Are there any performance concerns? Are there security issues? How are permissions managed? Is there a way to simplify this approach? Can this be moved to a shared function? Are there any downstream implications of this change? Ignore this concern for now and ensure this other part is handled first. How can this be revised to minimally solve the request?
* While reading code/doing code review:

  What is this function doing? How many callers does this function have? Does this consider the following edge case? Describe/make a diagram for the architecture of this feature. What is the user flow on this page? How are errors propagated? Are there any gaps here? Is there documentation, inline comments or tests around this, and do they match the functional logic?
* When writing docs, PR descriptions, summaries, etc.:

  What are the product implications of these changes for users? How does this affect downstream usage of this function? Update the architecture diagram to match these changes. Which engineers work on this code and should need to be informed about these changes?

File: ai/ai-at-tdd.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+Since our last Tidy Dev Day, one thing has changed dramatically: many participants will arrive with AI tools that can help them navigate an unfamiliar codebase, write a test, or suggest a possible fix. Before the day arrives, we want to be clear about how we hope participants will use them.
+
+## The point of the day is not just the pull request
+
+If our goal were simply to close issues, we wouldn’t need a venue, a barista, or a gong. Tidy Dev Day is about more than the diff:
+
+* **Demystifying contribution.** For many people, the barrier to contributing to open source isn’t ability—it’s the sense that contribution is something other people do. The day lets participants experience the full arc: claim a sticky note, work through the problem, submit a PR, get a review from a tidyverse developer, and smash the gong.
+
+* **Curated, accessible issues.** We select issues that are tractable in a day and genuinely useful, so beginners and veterans alike can find something at the right level of challenge. That challenge is a feature, not an inefficiency to optimize away.
```
Comment (hadley) [inline ai/ai-at-tdd.md]: Maybe it's worth explicitly acknowledging that these are also that issues that an agent is extremely likely to be successful with?

File: ai/ai-at-tdd.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+* **Learning and community.** Participants don’t need extensive development experience to come—just a GitHub account and basic git familiarity. Much of the value comes from working alongside other people: pairing with a stranger, asking a maintainer, “Why is it done this way?”, and cheering when someone else’s PR is merged.
+
+* **Building a community of maintainers.** The tidyverse’s long-term health depends on people who understand its code, not just those who can generate plausible fixes.
+
+These goals are about what happens in participants’ heads and between people, not just the diff. A fix its author can’t explain may close an issue, but it does not build that person’s ability, confidence, or connection to the project. Nor does it make the day easier for reviewers: every PR is reviewed by a real tidyverse developer, in the room, on the day. A pile of plausible-looking fixes that their authors cannot discuss moves the thinking onto a handful of reviewers.
```
Comment (hadley) [inline ai/ai-at-tdd.md]: Maybe reinforce this has always been true? Like even before AI it's faster (in the short-term) for us to just fix the issues.

File: ai/ai-at-tdd.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Where AI can help
+
+AI can be a tremendous accelerator when participants use it to *build* understanding rather than substitute for it. We encourage them to use it that way throughout the day:
+
+* **Understanding the issue.** Participants can paste in the issue thread and ask for a plain-language explanation. What is the reported behavior? What is expected? What R concepts are needed to understand the problem?
+
+* **Making a reprex.** A minimal reproducible example is a small, self-contained demonstration of a problem that others can run and inspect. AI can help participants strip a problem down to its essential pieces. Confirming that the reprex actually demonstrates the bug is also a quick check of their understanding.
```
Comment (hadley) [inline ai/ai-at-tdd.md]: Maybe it'll often get you 90% of the way, but you should spend some time personally seeing if you can reduce it further?

File: ai/ai-at-tdd.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+* **Getting oriented in an unfamiliar codebase.** “Where is `abort()` called from in this package?” “Walk me through what this S3 method does.” Questions like these can take a while to answer by grepping through an unfamiliar codebase; now they can take a minute—and leave participants knowing more.
```
Comment (hadley) [inline ai/ai-at-tdd.md]: Explain this traceback?

File: ai/ai-at-tdd.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+* **Learning and community.** Participants don’t need extensive development experience to come—just a GitHub account and basic git familiarity. Much of the value comes from working alongside other people: pairing with a stranger, asking a maintainer, “Why is it done this way?”, and cheering when someone else’s PR is merged.
+
+* **Building a community of maintainers.** The tidyverse’s long-term health depends on people who understand its code, not just those who can generate plausible fixes.
+
+These goals are about what happens in participants’ heads and between people, not just the diff. A fix its author can’t explain may close an issue, but it does not build that person’s ability, confidence, or connection to the project. Nor does it make the day easier for reviewers: every PR is reviewed by a real tidyverse developer, in the room, on the day. A pile of plausible-looking fixes that their authors cannot discuss moves the thinking onto a handful of reviewers.
+
```
Comment (hadley) [inline ai/ai-at-tdd.md:22]: Do we want to say something concretely like "Don't write a prompt where you just ask the agent to fix the problem"?
